src/main/java/edu/greenblitz/tobyDetermined/Nodesssss/MultiNodeSubsystem/AStarVertex.java

```diff
@@ -0,0 +1,170 @@
+        }
+        return new Pair<>(check,cost);
+    }
+
+    public static Boolean isVertexUsable(Vertex vertex) {
+        LinkedList<SystemsPos> otherSystemPositions = NodeBase.getAllSystemsPositions();
+        boolean isTherePossiblePosition = false;
+        for (int i = 0; i < otherSystemPositions.size() && !isTherePossiblePosition; i++) {
+            if (vertex.isPosFineForVertex(otherSystemPositions.get(i))) {
+                isTherePossiblePosition = true;
+            }
+        }
+        if (isTherePossiblePosition) {
+            return NodeBase.getNode(vertex.getOtherSystemPos()).getOtherSystemMustBeToOut().contains(vertex.getPos1())
+                    || NodeBase.getNode(vertex.getOtherSystemPos()).getOtherSystemMustBeToOut().isEmpty();
+        }
+        return false;
+    }
+
+    public static Double getCostOfOtherSystemPathAndAddToMap(Vertex vertex, HashMap<Vertex, Pair<LinkedList<SystemsPos>, Double>> pathMap) {
+        SystemsPos pos = vertex.getOtherSystemPos();
+        LinkedList<SystemsPos> list = vertex.mergeAndGetPos1OutAndPos2Enter();
+        double min = Double.MAX_VALUE;
+        LinkedList<SystemsPos> path = new LinkedList<>();
+        for (SystemsPos systemsPos : list) {
+            Pair<LinkedList<SystemsPos>, Double> pair = getPath(pos, systemsPos, vertex.getPos1());
+            if (pair.getSecond() < min) {
+                min = pair.getSecond();
+                path = pair.getFirst();
+            }
+        }
+        pathMap.put(vertex, new Pair<>(path, min));
+        vertex.setOtherSystemPos(path.get(path.size() - 1));
+        return min;
+    }
+
+    public static Pair<LinkedList<SystemsPos>, Double> returnPath(Vertex vertex, HashMap<Vertex, Vertex> parents, HashMap<Vertex, Pair<LinkedList<SystemsPos>, Double>> pathMap) {
+        LinkedList<SystemsPos> pathList = new LinkedList<>();
+        double totalCost = 0;
+        Vertex current = vertex;
+        pathList.addFirst(current.getPos2());
+        while (current != null) {
+            totalCost += current.getTimeCost();
+            if (!pathList.get(0).equals(current.getPos2()))
+                pathList.addFirst(current.getPos2());
+            if (pathMap.containsKey(current)) {
+                totalCost += pathMap.get(current).getSecond();
+                for (int i = 0; i < pathMap.get(current).getFirst().size(); i++) {
+                    if (!pathList.get(i).equals(pathMap.get(current).getFirst().get(i)))
+                        pathList.add(i, pathMap.get(current).getFirst().get(i));
+                }
+            }
+            if (!pathList.get(0).equals(current.getPos1()))
+                pathList.addFirst(current.getPos1());
+            current = parents.get(current);
+        }
+        return new Pair<>(pathList, totalCost);
+    }
+
+    public static Pair<LinkedList<SystemsPos>, Double> getPath(SystemsPos start, SystemsPos end, SystemsPos secondSystemPos) {
```

Comment:
state the difference between get, return and return final path in the name or get good